### ISSUE TITLE
Question: Can the "Recent Folders" listing be navigated (up/down) from keyboard?

### ISSUE THREAD

**Issue by jaysmithgit** [RealRaven2000/QuickFolders#561]
I'm sorry about having to ask this, but my goal is to find out:

1) If what I am asking about is possible on the versions of QF and TB that I *currently* have (and I just can't figure out how to do it)?
2) If it is possible to do in later versions?
3) And if not possible now or in later versions, can it be submitted as a feature suggestion?

QF Pro 5.9pre33 ... on TB 78.7.1 ... on Ubuntu Linux 20.04
(Yes, I know I have to upgrade -- everything -- but I am still locked in at the moment for several reasons -- but an upgrade is finally coming soon.)

To be more efficient, I am trying to do more using *only* the keyboard, but I am stumped on this action...

I have defined custom keystrokes (again, I am  using QF Pro) to COPY (c) and to MOVE (s) emails into pre-existing folders.

Currently when I do a "c" or an "s" the focus goes to the "Recent Folders" box where I currently seem (????) to *only* have the choices of *either* 

a) ... selecting (i.e. Enter key) the folder name that is visible (only one is visible -- the most recent folder accessed by QF) and the email gets copied or moved to that folder *or* 

b) ... TYPING the name (or part of) the desired folder name in the Recent Folders box, navigate up/down through the presented list of possibilities an then select one (Enter key) so that the email gets copied or moved to that folder

My "problem" is that as I am going through the Inbox and filing emails, I might have several that go into the a varied, but small, group of folders. But I need to read them one at a time and then copy/move them to their correct folders. 

For example, there might be 10 emails that I am reviewing them as they are in date order, and after reviewing each one, I will copy/move it, INTO *ONE* THESE FOLDERS, but the folder varies each time but is eventually repeated:

george
fred
alex
bob
fred
george
amy
bob
... etc

At this time, on the QF/TB versions I currently have, the only way I can accomplish that USING *ONLY* THE KEYBOARD is to start *TYPING* EACH NAME FOR *EACH* EMAIL.

It does *not* allow me to just type "c" or "s" (my assigned keystrokes) and see a LIST of recent folders in the Recent Folders box/list/widget. And the keyboard up/down arrows don't do anything -- there is no list showing in which I can move around.

The only way I can do something *like* what I am describing is to use my MOUSE to drag an email to hover over the Recent Folders list ... when I do use the mouse, then a whole list of recent folders DOES APPEAR and I can drag the email to the folder into which want to copy/move the email.

??? Is there [either back in QF 5.9pre33, or now] a keystroke(s) command that will allow me to see a LIST of the Recent Folders and then move up/down in the list, select a folder and move/copy into the selected folder?

When I type a "g" for "go into a folder", the results are a little different.  I DO then see a list, but it is a list of the search results from the last time I typed a partial folder name in the Recent Folders box. 

??? So, similarly, if just want to *just using the keyboard* GO into a folder (not copy or move an email), how can I see (and select from) the entire Recent Folders listing (that I would see if I were using a mouse to click on Recent Folders)?

I feel like I must be missing something or just can't find the correct Help page, etc.  QF works so wonderfully in every other way.

P.S. If there was a Nobel Prize for s/w quality and responsive to users and keeping up with changes (i.e. TB makes changes, so QF has to change), our "QF Axel" would be sure to receive it!  :-)

Jay

**Comment by RealRaven2000** [comment]
There is good news and bad news:

Good news first: I am actually working on keyboard accessibility right now, see all issues marked with the new label "accessibility":
#560 #551 #550 #548 and the ongoing "collection" bug for keyboard navigation #555

these changes all go into the current version of QuickFolders

Bad news (for you, if you're still stuck on 78): I cannot support really old versions. I am trying my best to make Add-ons backwards compatible with multiple ESR versions but this gets harder every year. I had to make a hard cut 2 years ago when Thunderbird 115 came out as it dramatically rewrote the complete front page (3pane area) and a lot of other tabs as well - that's when I made a completely new branch " ESR 115" and increased my major version from 5=>6. That number is significant, it means there's no way back!

The current version have just released (6.10.1) is compatible from 115 - 137. And Thunderbird this month switched to monthly releases (same as Firefox was doing for years) as a default. You can still download the ESR versions, but Release will be default, and I will stay compatible with that for the future in order not disrupt my users.

Now as regards seeing recent folders using the keyboard, use the quickJump shortcut (Shift+J is the default) and then type `=` in the search field (overwriting what's there)

![Image](https://github.com/user-attachments/assets/75a9b065-bf33-40d5-908e-122e509af5d9)

You can check on your own version whether this shortcut already exists, by clicking the blue question mark to the right of the search box:

![Image](https://github.com/user-attachments/assets/55dd8085-ddaa-44a1-bf7f-4846860c6f34)

This screenshot was made with Thunderbird 91 (one of my oldest test profiles) with QuickFolders 5.18, and it seems to support the `=` shortcut in the search field. To me, Thunderbird91 is already a very old version.

**Comment by jaysmithgit** [comment]
Wow, thanks.

And I appreciate the opportunity to discover a long-standing spelling bug in the code (which you indicate you have now fixed). Actually, I am that guy that finds old bugs and errors. My record is an error that had been staring all users of an annual reference book that had been there for more than 50 years.

Unfortunately, I am stuck on this old version for a little while longer. But we are working on it... BTW, my version is before the "blue question mark". :-) And doing the "J' does not do anything in my version.  But that's okay because...

Doing my normal single-letter commands for (go, move, copy) does get me to the Recent Folders box (as I previously stated) but then doing a "=" *DOES WORK* to get the full recent folders list ... and it *IS* navigable with the up/down arrows.

THANK YOU, that "=" will be a huge time saver for me.

??? Where is the "=" documented (back then), or better, is it documented now? 

Jay

**Comment by RealRaven2000** [comment]
> ??? Where is the "=" documented (back then), or better, is it documented now?

the big question, when was the feature added officially, I think it may have been later...

check out https://quickfolders.org/version.html#5.9.1

> [[issue 242]](https://github.com/RealRaven2000/QuickFolders/issues/242)
>
> Use '=' key for recalling folders from recent quickMove » quickJump operations
>
> When using the quickMove or quickJump search box you can now access folders of its previous jump » move operations by typing "=" as the first character.

I am not sure whether I mentioned it on the QuickFolders home page, except in the "Highlights" box on the top right. Generally I did most of the documentation in the (full localized) box that pops up on the blue question mark, here is the full English text (hope a screenshot is ok).

![Image](https://github.com/user-attachments/assets/c620bbfb-6820-426a-bd8e-0895c521c8be)

there is a lot of info in here. **_Guess what I need is a full chapter on the quickfolders.org site_**? There is some description on the "premium" features page:

https://quickfolders.org/premium.html#quickmove

The shortcut is not mentioned in the version history for 4.* let's check:

https://quickfolders.org/version-68.html

ok I can only find when **quickMove** was added in [[FR 25825]](https://quickfolders.org/bugzilla/bugs/show_bug.cgi@id=25825) 

https://quickfolders.org/version-68.html#4.0.3

but I cannot find a mention of the "=" shortcut. I am not sure if the code base was already on github, but I guess it was on a cvs repository on the now defunct mozdev.org server. I don't think the version history was imported into then google code (which was later migrated to github). Historically QF is a long project, at least by IT time standards.

Btw, if you configure the Recent Folders Tab in the main toolbar can you use cursor down after clicking on it?

![Image](https://github.com/user-attachments/assets/eea479a1-2f99-4dd9-8b31-057cd870d030)

As I am working on accessibility I just fixed another bug in 6.10.1 (awaiting review since yesterday) see #562. I might do a quick emergency release and bump up to 6.10.2 before I hit the gym this morning...

**Comment by RealRaven2000** [comment]
Just checked, the Recent folders dropdown can be navigated with keyboard (up/down) at least since the recently implemented accessibility changes.
